.chromatic/custom-addons/chromatic/index.js

```diff
@@ -25,41 +27,78 @@ export const withChromaticProvider = makeDecorator({
       height = options.height;
     }
 
-    let expressTheme = colorSchemes.length === 1 ? expressThemes[colorSchemes[0]] : expressThemes.light;
-    let expressColorScheme = colorSchemes.length === 1 ? colorSchemes[0].replace(/est$/, '') : 'light';
-    let expressScale = scalesToRender.length === 1 ? scalesToRender[0] : 'medium';
-    let expressLocale = selectedLocales.length === 1 ? selectedLocales[0] : 'en-US';
-
-    // do not add a top level provider, each provider variant needs to be independent so that we don't have RTL/LTR styles that interfere with each other
-    return (
-      <DisableAnimations disableAnimations={options.disableAnimations}>
-        <div style={{display: 'flex', flexDirection: 'column', height, minHeight}}>
-            {colorSchemes.map(colorScheme =>
-              scalesToRender.map(scale =>
-                (colorScheme === 'light' ? selectedLocales : ['en-US']).map(locale =>
-                  <Provider key={`${colorScheme}_${scale}_${locale}`} theme={themes[colorScheme]} colorScheme={colorScheme.replace(/est$/, '')} scale={scale} locale={locale} typekitId="pbi5ojv">
-                    <View margin="size-100">
-                      <h1 style={{margin: 0, padding: 0}}>{`${colorScheme}, ${scale}, ${locale}`}</h1>
-                      {getStory(context)}
-                    </View>
-                  </Provider>
-                )
-              )
-            )}
-            {options.express !== false &&
-              <Provider key="express" theme={expressTheme} colorScheme={expressColorScheme} scale={expressScale} locale={expressLocale} typekitId="pbi5ojv">
-                <View margin="size-100">
-                  <h1 style={{margin: 0, padding: 0}}>express, {expressColorScheme}, {expressScale}, {expressLocale}</h1>
-                  {getStory(context)}
-                </View>
-              </Provider>
-            }
-        </div>
-      </DisableAnimations>
-    )
+    if (context.title.includes('S2/')) {
+      return <RenderS2 getStory={getStory} context={context} options={options} selectedLocales={selectedLocales} height={height} minHeight={minHeight} />
+    } else {
+      return <RenderV3 getStory={getStory} context={context} options={options} selectedLocales={selectedLocales} height={height} minHeight={minHeight} />
+    }
   }
 });
 
+function RenderS2({getStory, context, options, selectedLocales, height, minHeight}) {
```

Comment:
Could you make this into a shared component that .chromatic-fc and .chromatic both use? I assume there is no way you could make this general enough to make it not specific to s2 or RSPv3, the s2 or RSP3 parts are passed in, or toggles between them based on a property? It's just a big ugly block of code getting repeated. It's probably not worth the effort to refactor it this way.

As part of this, I was wondering if making it a component would move the style macro import out of this file and into another file where it might work.

Reply:
for now I'll leave it as its own block until I figure out the full set of differences when I add the powerset prop combination stories for s2, but yeah it would be nice to unify this